### DISCUSSION TITLE
No Astro plugin for Jetbrains Webstorm IDE

### DISCUSSION THREAD

**Discussion by rehanvdm** [withastro/roadmap#226] [Proposal]
So are we just going to ignore the fact that many developers work on Webstorm IDE? 

The docs only acknowledges VS Code as an editor? https://docs.astro.build/en/editor-setup/

I was really excited for astro, but it doesn't seem ready for the spotlight.

**Comment by natemoo-re** [comment]
Hey @rehanvdm, thanks for starting a discussion.

We would love to support Webstorm IDE! Unfortunately, it does not support language servers, so creating an Astro extension would mean we'd have to write and maintain an entirely separate extension in a different language than our existing codebase. We currently don't have the bandwidth for this, but hopefully we will in the future.

We are definitely keeping an eye on JetBrains' Fleet which _will_ support language servers—our currently available tooling will be able to run there without any problem.

**Comment by rehanvdm** [comment]
Okay I understand, sorry for my attitude in the description 😅 I just really did not look forward to doing SSG and all the cool stuff Astro is doing myself with another tool and it poured into the ticket.

FYI these are tickets I saw yesterday about Astro and Intelli J support 
- https://youtrack.jetbrains.com/issue/WEB-52015
- https://youtrack.jetbrains.com/issue/IDEA-291881/Frontmatter-Support

I just hope that this does not become a finger-pointing game; Astro - "we can not do it because Jetbrains does not support it" or Jetbrains - "it is up to Astro to implement this"

Appreciate the response. There aren't a lot of SSG tools for Vue 3 out there, Astro is one of them and I hope to work with it soon. In the interim, I choose to vite-ssg, but it has a much steeper learning curve, had to deep dive to sort issues and the community + support is not nearly as big, but it will do for now. 

Thanks again.

**Comment by privetkuku** [comment]
Hey @natemoo-re, how do you do?
So, Fleet is out for everybody, but I see that it does not support Plugins like VS Code, meaning we should wait for Astro support from JetBrains?

**Reply by Princesseuh** [reply]
Once Fleet add support for plugins, we'll develop an Astro plugin for it. At the moment there's nothing we can do for Fleet unfortunately

**Comment by rehanvdm** [comment]
@natemoo-re Is it supported yet 😬?

**Reply by Princesseuh** [reply]
The Webstorm team is working on it! It's part of their Q1 roadmap

**Reply by rehanvdm** [reply]
That sounds awesome, excited 💪

**Reply by FredKSchott** [reply]
https://blog.jetbrains.com/webstorm/2022/12/webstorm-2023-1-roadmap/

**Reply by ShashiSrinath** [reply]
Astro plugin is available for the EAP version of the Jetbrains IDE's.
Works perfectly on Intellij but im facing some issues on webstorm.

**Comment by natemoo-re** [comment]
WebStorm recently launched official support for Astro, check out [our blog post](https://astro.build/blog/astro-webstorm-support/) for more info!
